src/core/model_repository_manager.cc

```diff
@@ -842,17 +842,20 @@ ModelRepositoryManager::Create(
   if (polling_enabled) {
     RETURN_IF_ERROR(
         local_manager->Poll(&added, &deleted, &modified, &unmodified));
-  }
-  if (!deleted.empty() || !modified.empty() || !unmodified.empty()) {
-    return Status(
-        RequestStatusCode::INTERNAL,
-        "Unexpected initial state for model repository");
-  }
 
-  RETURN_IF_ERROR(local_manager->Update(added, deleted, modified));
+    if (!deleted.empty() || !modified.empty() || !unmodified.empty()) {
+      return Status(
+          RequestStatusCode::INTERNAL,
+          "Unexpected initial state for model repository");
+    }
 
-  // model loading / unloading error will be printed but ignored
-  local_manager->LoadModelByDependency();
+    RETURN_IF_ERROR(local_manager->Update(added, deleted, modified));
+
+    // [TODO] fix this. On init, should return loading error and let
```

Comment:
Is this DLIS-506?

Reply:
Yes.

Reply:
I think it is ok to but the DLIS-506 in the comment. It doesn't mean anything externally but it helps us remember.